src/sweeps/config/schema.json

```diff
@@ -403,6 +553,16 @@
         "min"
       ],
       "properties": {
+        "nested": {
+          "type": "boolean",
+          "description": "Whether this parameter represents a nested parameter in the Run config.",
+          "default": false
+        },
+        "nest_delimiter": {
+          "type": "string",
+          "description": "Delimiter to use when unnesting parameter into a Run config.",
+          "default": "."
+        },
```

Comment:
this interface adds boilerplate to every parameter definition in the schema, and it adds lots of corner cases that we have to be sure to check and validate, which is the primary source of sweeps bugs these days on sentry. once this becomes a part of our API we would have to support it and it would be hard to move away.

we should move to a schema where it is hard as possible to construct invalid data that validates against the schema. a special nested parameter type, the shape of which mirrors the actual config structure, could work here. then we could also get rid of `nest_delimiter`...